src/main/c/cext/ruby.c

```diff
@@ -976,11 +976,25 @@ double rb_float_value(VALUE value) {
 // String
 
 char *RSTRING_PTR_IMPL(VALUE string) {
-  return (char *)truffle_invoke(RUBY_CEXT, "RSTRING_PTR", string);
+  char* ret = truffle_invoke(RUBY_CEXT, "RSTRING_PTR", string);
+
+  // We start off treating RStringPtr as if it weren't a pointer for interop purposes. This is so Sulong doesn't try
+  // to convert it to an actual char* when returning from `truffle_invoke`. Once we have a handle to the real RStringPtr
+  // object, we can instruct it to start acting like a pointer, which is necessary for pointer address comparisons.
+  truffle_invoke_i(ret, "act_like_pointer=", Qtrue);
```

Comment:
Why `_i` here? Maybe `_b` would be better?

Reply:
I looked for `_b` and didn't see it. Evidently I just overlooked it. Thanks.